### PR TITLE
OCT-100: Add Rector

### PR DESCRIPTION
**Pull request by tseho**
**Description (for Contributor and Core Developer)**

Rector can upgrade existing code to newer standards. It can refactor automatically.
Adding it to our lint back workflow will guarantee that old syntaxes are not re-introduced.

Example of rector's output:
```
1) components/catalogs/back/src/Infrastructure/TemporaryEnrichmentBridge/SearchAfterSizeUuidResultCursorFactory.php:21

    ---------- begin diff ----------
@@ @@
  */
 class SearchAfterSizeUuidResultCursorFactory implements CursorFactoryInterface
 {
-    /** @var Client */
-    private $esClient;
-
-    public function __construct(Client $esClient)
+    public function __construct(private Client $esClient)
     {
-        $this->esClient = $esClient;
     }

     /**
@@ @@
      * @param mixed $queryBuilder
      * @param array<array-key, mixed> $options
      */
-    public function createCursor($queryBuilder, array $options = [])
+    public function createCursor($queryBuilder, array $options = []): \Akeneo\Pim\Enrichment\Bundle\Elasticsearch\IdentifierResultCursor
     {
         /** @var array{_source: array<string>, sort?: array<string>, size: int} $queryBuilder */
    ----------- end diff -----------

Applied rules:
 * ClassPropertyAssignToConstructorPromotionRector (https://wiki.php.net/rfc/constructor_promotion https://github.com/php/php-src/pull/5291)
 * ReturnTypeFromReturnNewRector
 ```
 
 Checkout https://github.com/akeneo/pim-community-dev/pull/17648 for the full list of changes.

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/back/rector.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        LevelSetList::UP_TO_PHP_80,
+    ]);
+
+    $rectorConfig->rules([
+        MakeInheritedMethodVisibilitySameAsParentRector::class,
+    ]);
+
+    $rectorConfig->skip([
+        FlipTypeControlToUseExclusiveTypeRector::class,
+        ExplicitBoolCompareRector::class,
```
Comment (masacc) [inline components/catalogs/back/rector.php:33]: :cry:

**Review by masacc — APPROVED** [review]

File: components/catalogs/back/rector.php (hunk 1 above)
Comment (tseho) [reply components/catalogs/back/rector.php:33]: Which one ? `ExplicitBoolCompareRector` ?

**Review by RiffVentura — APPROVED** [review]

File: components/catalogs/back/rector.php (hunk 1 above)
Comment (masacc) [reply components/catalogs/back/rector.php:33]: yep, in PHP I like to be explicit in comparisons

Comment (tseho) [reply components/catalogs/back/rector.php:33]: Ok, I'm going to try to fix the 2 dirty ones that were preventing me to use it.

**Pull request merged**